packages/foam-vscode/src/features/open-daily-note.ts

```diff
@@ -63,7 +63,7 @@ async function createDailyNoteIfNotExists(
     return false;
   }
 
-  createDailyNoteDirectoryIfNotExists(dailyNotePath);
+  await createDailyNoteDirectoryIfNotExists(dailyNotePath);
```

Comment:
I think there was an eslint plugin/rule to help with these types of oversights. Such a killer to debug these. Well done for picking it up. 🙏

Reply:
https://eslint.org/docs/rules/require-await

Reply:
I went ahead and added the require-await to this pull request. I then cleaned up the warnings for this file specifically. So ya know, require-await wouldn't have found this particular bug because there are already other awaits inside of that function, but it still seems like a handy warning!